docs/tutorial/bigquery.md

```diff
@@ -0,0 +1,153 @@
+# Connect Streamlit to Google BigQuery
+
+## Introduction
+
+This guide explains how to securely access a BigQuery database from Streamlit sharing or Streamlit for Teams. It uses the [google-cloud-bigquery](https://googleapis.dev/python/bigquery/latest/index.html) library and Streamlit's [secrets management](../deploy_streamlit_app.html#secrets-management).
+
+
+## Create a BigQuery database
+
+If you already have a database that you want to use, feel free to [skip this step](#enable-the-bigquery-api).
+
+For this example, we will simply use one of the [sample datasets](https://cloud.google.com/bigquery/public-data#sample_tables) from BigQuery (namely the `shakespeare` table). If you want to create a new dataset instead, follow [Google's quickstart guide](https://cloud.google.com/bigquery/docs/quickstarts/quickstart-web-ui).
+
+## Enable the BigQuery API
+
+Programmatic access to BigQuery is controlled through [Google Cloud Platform](https://cloud.google.com). Create an account or sign in and head over to the [APIs & Services dashboard](https://console.cloud.google.com/apis/dashboard) (select or create a project if asked). As shown below, search for the BigQuery API and enable it:
+
+```eval_rst
+.. tip:: To enlarge the images, simply click on them.
+
+.. thumbnail:: ../media/databases/big-query-1.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-2.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-3.png
+   :width: 32%
+```
+
+## Create a service account & key file
+
+To use the BigQuery API from the Streamlit Cloud, you need a Google Cloud Platform service account (a special account type for programmatic data access). Go to the [Service Accounts](https://console.cloud.google.com/iam-admin/serviceaccounts) page and create an account with the **Viewer** permission (this will let the account access data but not change it):
+
+```eval_rst
+.. thumbnail:: ../media/databases/big-query-4.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-5.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-6.png
+   :width: 32%
+
+.. note:: If the button **CREATE SERVICE ACCOUNT** is gray, you don't have the correct permissions. Ask the admin of your Google Cloud project for help.
+```
+
+After clicking **DONE**, you should be back on the service accounts overview. Create a JSON key file for the new account and download it:
+
+```eval_rst
+.. thumbnail:: ../media/databases/big-query-7.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-8.png
+   :width: 32%
+
+.. thumbnail:: ../media/databases/big-query-9.png
+   :width: 32%
+```
+
+```eval_rst
+.. important:: Store the key file in a safe location and don't share it with anyone.
```

Comment:
I'm leaning towards removing this callout altogether. Google's IAM [documentation](https://cloud.google.com/iam/docs/creating-managing-service-account-keys#iam-service-account-keys-create-console) does not specify what constitutes a safe/secure location either. They say "*Make sure to store the key file securely, because it can be used to authenticate as your service account. You can move and rename this file however you would like.*"